contracts/src/challenge/SymChallenge.sol

```diff
@@ -86,23 +86,46 @@ contract SymChallenge is ChallengeBase, ISymChallenge {
         daProvider = _daProvider;
         resultReceiver = _resultReceiver;
         startStateHash = _startStateHash;
-        endStateHash = _endStateHash;
 
         turn = Turn.Defender;
         lastMoveBlock = block.number;
         defenderTimeLeft = challengePeriod;
         challengerTimeLeft = challengePeriod;
     }
 
-    // TODO: use minimum of challenger/defender's proposed numSteps.
-    function initializeChallengeLength(uint256 _numSteps) external override onlyOnTurn {
+    function proposeChallenge(bytes32 _endStateHash, uint256 _numSteps) external {
+        // This can be run before turn checking and probably saves gas
         if (bisectionHash != 0) {
             revert AlreadyInitialized();
         }
         require(_numSteps > 0, "INVALID_NUM_STEPS");
-        bisectionHash = ChallengeLib.initialBisectionHash(startStateHash, endStateHash, _numSteps);
-        // TODO: consider emitting a different event?
-        emit Bisected(bisectionHash, 0, _numSteps);
+
+        // Defender proposes `numSteps` and `endStateHash` first
+        if (turn == Turn.Defender && msg.sender == defender) {
```

Comment:
should we use the `onlyOnTurn` modifier in the function definition? then you can just check
`if (turn == Turn.Defender)` and also don't need to do the reverts below